### PR TITLE
FIX: Set solder ball fix with negative diameter

### PR DESCRIPTION
**Pull request by svandenb-dev**
For some cases EDB api is returning negative pad dimension causing solder balls settings failure.

closing #736

### CODE REVIEW

File: src/pyedb/dotnet/edb_core/components.py [hunk 1]
```diff
@@ -871,7 +871,6 @@ def create_port_on_pins(self, refdes, pins, reference_pins, impedance=50.0, port
                 elif "-" in ref_pin_name and ref_pin_name.split("-")[1] in refdes_pins:
                     ref_cmp_pins.append(refdes_pins[ref_pin_name.split("-")[1]])
             if not ref_cmp_pins:
-                self._logger.warning("No reference pins found during port creation. Port is not defined.")
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/components.py:874]: Why do you remove this warning ? Can you make it a debug warning if you don't want people to see it ?

File: src/pyedb/dotnet/edb_core/components.py [hunk 2]
```diff
@@ -2071,8 +2070,8 @@ def set_solder_ball(
             pin1 = list(cmp.pins.values())[0].pin
             pin_layers = pin1.GetPadstackDef().GetData().GetLayerNames()
             pad_params = self._padstack.get_pad_parameters(pin=pin1, layername=pin_layers[0], pad_type=0)
-            _sb_diam = min([self._get_edb_value(val).ToDouble() for val in pad_params[1]])
-            sball_diam = _sb_diam
+            _sb_diam = abs(min([self._get_edb_value(val).ToDouble() for val in pad_params[1]]))
+            sball_diam = 0.8 * _sb_diam
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/components.py:2074]: Why do you multiply by 0.8 ?

**Review by SMoraisAnsys — COMMENTED** [review]
Are you sure that returning the absolute value is enough ? Couldn't we have cases like [-0.6, 0.3] where 0.3 should be the value returned and this would return 0.6 ?

File: src/pyedb/dotnet/edb_core/components.py (hunk 2 above)
Comment (svandenb-dev) [reply src/pyedb/dotnet/edb_core/components.py:2074]: this is an arbitrary value based on experiment. When users export to 3D modeler having the same value cause trouble with meshing. Using a smaller value prevent this issue

**Pull request merged**